vignettes/mapping.Rmd

```diff
@@ -0,0 +1,239 @@
+
+```{r}
+wind <- griddap(
+  'erdQMwindmday', time = c('2016-11-16', '2017-01-16'),
+  latitude = c(30, 50), longitude = c(210, 240),
+  fields = 'x_wind'
+)
+unique(wind$data$time)
+```
+
+When faced with multiple time periods, and `animate = FALSE` (the default), the `time` argument is used to reduce multiple grids (i.e., raster bricks) to a single grid (i.e., a single raster layer). You can pass any R function to the `time` argument, but when `animate = FALSE`, you should take care to ensure the function returns a single value. The default uses the `mean()` function so that each cell represents the average (in this case amongst three time points), but we could easily set this to `var()` to get the variance for each cell:
+
+```{r, fig.align='default'}
+# add space for a title
+par(mar = c(0, 0, 1, 0))
+add_griddap(
+  plotdap("base", crs = alaska, mapTitle = "Mean wind speed"), 
+  wind, ~x_wind
+)
+add_griddap(
+  plotdap("base", crs = alaska, mapTitle = "Variance of wind speed"), 
+  wind, ~x_wind, time = var
+)
+```
+
+If you just want to animate the raw grid values over time (instead of computing summary statistic(s) for each cell), you can set `time` to the `identity` function, then set `animate = TRUE`:
+
+```{r, eval = FALSE}
+add_griddap(
+  plotdap(crs = "+proj=robin"), 
+  wind, ~x_wind, 
+  time = identity, 
+  animate = TRUE
+)
+```
+
+<div align = "center">
+  <a href="http://i.imgur.com/aev0aZt">
+    <img src = "http://i.imgur.com/aev0aZt.gif" height = "500" width = "700" />
+  </a>
+</div>
+
+
+## Combining tables/grids
+
+The `plotdap()` interface is designed so that you can plot both tables and grids at once (although I don't recommend trying to plot more than one `griddap()` or `tabledap()` object at once). It's also designed to work well with the `%>%` operator from the **magrittr** package so that code can be expressed in a left-to-right (rather than inside-out) fashion:
+
+```{r, fig.width=7}
+plotdap("base") %>%
+  add_griddap(murSST, ~analysed_sst) %>%
+  add_tabledap(sardines, ~subsample_count)
+```
+
+## Modifying `ggplotdap` objects
+
+By this point, you might have noticed some subtle differences in the defaults of `plotdap("ggplot2")` versus `plotdap("base")`. The default base version is left intentially minimal as it is often much harder (or impossible) to modify/remove elements from a base graphic once it's drawn. However, this is fairly easy using the **ggplot2** approach (assuming you know a bit of **ggplot2**), thanks to the `add_ggplot()` function (think of it like the `+` operator in **ggplot2**). 
+
+```{r, fig.width=7}
+library(ggplot2)
+plotdap(crs = "+proj=robin") %>%
```

Comment:
I suppose it might not be worth the headache -- just wanted to show you can add themes from other packages